crates/relayer/src/config.rs

```diff
@@ -509,4 +561,43 @@ mod tests {
         let mut buffer = Vec::new();
         store_writer(&config, &mut buffer).unwrap();
     }
+
+    #[test]
+    fn gas_price_try_from() {
+        let gp_original = GasPrice::new(10.0, "atom".to_owned());
+
+        let gp_raw = gp_original.to_string();
+        let gp: GasPrice = gp_raw
+            .try_into()
+            .expect("could not parse String into GasPrice");
+
+        assert_eq!(gp, gp_original);
+    }
+
+    #[test]
+    fn parse_multiple_gas_prices() {
+        let gas_prices = "0.25token1;0.0001token2";
+        let parsed = parse_gas_prices(gas_prices.to_string());
+
+        let expected = vec![
+            GasPrice {
+                price: 0.25,
+                denom: "token1".to_owned(),
+            },
+            GasPrice {
+                price: 0.0001,
+                denom: "token2".to_owned(),
+            },
+        ];
+
+        assert_eq!(expected, parsed);
+    }
+
+    #[test]
+    fn parse_empty_gas_price() {
+        let empty_price = "";
+        let parsed = parse_gas_prices(empty_price.to_string());
+
+        assert_eq!(parsed, vec![]);
+    }
```

Comment:
Great stuff! Let's add a couple tests for when the `gas_price` string contains malformed/invalid data?

Reply:
[8bc1df9](https://github.com/informalsystems/hermes/pull/2873/commits/8bc1df9b1fedeb490ce6e5576f69e7db3762202f)